webapp/app/modules/userAuth/userAuth.sagas.js

```diff
@@ -0,0 +1,118 @@
+import { takeLatest, put, select, all, take } from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { replace } from 'react-router-redux';
+import reportError from 'report-error';
+import firebase from 'firebase';
+
+import { UserAuthTypes, UserAuthActions } from './userAuth.redux';
+import { selectLoggedUser } from '../users/users.selectors';
+import { selectUser } from './userAuth.selectors';
+import { UsersActions } from '../users/users.redux';
+import { selectLocationState } from '../router/router.selectors';
+
+
+const provider = new firebase.auth.FacebookAuthProvider();
+
+function* signInAnonymously() {
+  try {
+    yield firebase.auth().signInAnonymouslyAndRetrieveData();
+  } catch (error) {
+    reportError(error);
+  }
+}
+
+function* signInViaFacebook() {
+  try {
+    const { user: { uid, email, displayName, photoURL: profilePhoto, isAnonymous } } =
+      yield firebase.auth().signInWithPopup(provider);
+    yield put(UserAuthActions.setUserData(uid, isAnonymous));
+    yield put(UserAuthActions.checkUserAccount({ uid, email, displayName, profilePhoto }));
+  } catch (error) {
+    reportError(error);
+  }
+}
+
+function* checkUserAccount({ user }) {
```

Comment:
nie do końca rozumiem co robi ta funkcja i nazwa też nie pomaga w domyśleniu się. Możesz dać znać o co tutaj chodzi?

Reply:
Sprawdza czy taki użytkownik już istnieje w bazie, jeśli nie to go tworzy.

Reply:
no to może `checkIfUserAccountExists` ?

Reply:
`checkIfUserAccountExistsAndIfItDoesNotCreateIt` ? :D